src/sparseml/optim/helpers.py

```diff
@@ -230,18 +230,100 @@ def evaluate_recipe_yaml_str_equations(recipe_yaml_str: str) -> str:
         # yaml string does not create a dict, return original string
         return recipe_yaml_str
 
-    # validate and load remaining variables
-    container, variables, non_val_variables = _evaluate_recipe_variables(container)
+    # check whether the recipe is a stage recipe of not
+    if check_if_staged_recipe(container):
+        container = _evaluate_staged_recipe_yaml_str_equations(container)
+
+    else:
+        container, variables, non_val_variables = _evaluate_container_variables(
+            container
+        )
 
-    # update values nested in modifier lists based on the variables
-    for key, val in container.items():
-        if "modifiers" not in key:
-            continue
-        container[key] = _maybe_evaluate_yaml_object(val, variables, non_val_variables)
+        # update values nested in modifier lists based on the variables
+        for key, val in container.items():
+            if "modifiers" not in key:
+                continue
+            container[key] = _maybe_evaluate_yaml_object(
+                val, variables, non_val_variables
+            )
 
     return rewrite_recipe_yaml_string_with_classes(container)
 
 
+def check_if_staged_recipe(container: dict) -> bool:
+    """
+    Check whether container pertains to a staged recipe.
+    Such a "staged container" fulfills two conditions:
+    - no top level key in container contains "modifiers" in its name
+    - a stage should map to a dict that has at least one key with
+      "modifiers" in its name
+    :param container: a container generated from a YAML string of SparseML recipe
+    :return: True if stage recipe, False if normal recipe
```

Comment:
I don't like the word `normal` for a recipe without stages, Could we simplify this by saying something like:
`True if staged recipe, False otherwise`